### PR TITLE
[QAA-346][Detox] First Speculos Ledger Sync test

### PR DESCRIPTION
**Pull request by abdurrahman-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached. (no need)
- [X] **Covered by automatic tests.** 
- [X] **Impact of the changes:** 
  - LLM Speculos tests

### 📝 Description

_First Ledger Sync tests for LLM similar to [LLD](https://github.com/LedgerHQ/ledger-live/pull/8448)_

- Creation of the trustchain + push of 2 accounts to the cloud with the CLI
- Connection to the trustchain and checking that the accounts are visible on LLM
- Trustchain destruction on LLM

### ❓ Context

- [Speculos CI run](https://github.com/LedgerHQ/ledger-live/actions/runs/12080896108/job/33688996254)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-mobile/e2e/specs/speculos/ledgerSync/ledgerSync.spec.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    await app.ledgerSync.selectUseLedger();
+    await app.common.selectKnownDevice();
+    await activateLedgerSync();
+    await app.ledgerSync.expectLedgerSyncSuccessPage();
+    await app.ledgerSync.closeActivationSuccessPage();
+    await app.accounts.openViaDeeplink();
+    await app.accounts.expectAccountsNumber(2);
+    await goToLedgerSync();
+    await app.ledgerSync.openDeleteSync();
+    await device.disableSynchronization();
```
[the remark this answers is not in the document]
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/specs/speculos/ledgerSync/ledgerSync.spec.ts:93]: > 09:50:04.731 detox[82580] i The app has not responded to the network requests below:
  (id = 27) invoke: {"type":"action","action":"tap","atIndex":0,"predicate":{"type":"id","value":"delete-trustchain","isRegex":false}}
>
>That might be the reason why the test "Ledger Sync Accounts Synchronize one instance then delete the backup" has timed out.
>
>09:50:04.733 detox[82580] i Ledger Sync Accounts: Synchronize one instance then delete the backup [FAIL]

**Review by KVNLS — DISMISSED** [review]
That's great to see e2e tests for Ledger Sync!

File: apps/ledger-live-mobile/e2e/page/accounts/accounts.page.ts [hunk 2]
```diff
@@ -20,4 +27,9 @@ export default class AccountsPage {
   async addAccount() {
     await tapByElement(this.addAccountButton());
   }
+
+  @Step("Expect accounts number")
```
Comment (bharamboure-ledger) [inline apps/ledger-live-mobile/e2e/page/accounts/accounts.page.ts:31]: might make sense to log parameter in @step annotation

**Review by bharamboure-ledger — DISMISSED** [review]

File: apps/ledger-live-mobile/e2e/page/accounts/accounts.page.ts (hunk 2 above)
Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/e2e/page/accounts/accounts.page.ts:31]: Parameters are automatically logs
Screenshot 2024-12-03 at 08 21 59

**Review by bharamboure-ledger — APPROVED** [review]

**Pull request merged**